vizro-ai/src/vizro_ai/_vizro_ai.py

```diff
@@ -76,18 +78,29 @@ def _run_plot_tasks(
 
         pass_validation = validated_code_dict.get("debug_status")
         code_string = validated_code_dict.get("code_string")
-        business_insights, code_explanation = None, None
 
-        if explain and pass_validation:
-            business_insights, code_explanation = self._lazy_get_component(GetCodeExplanation).run(
-                chain_input=user_input, code_snippet=code_string
+        if not pass_validation:
+            raise DebugFailure(
+                "Chart creation failed. Retry debugging has reached maximum limit. Try to rephrase the prompt, "
+                "or try to select a different model. Fallout response is provided: \n\n" + code_string
             )
-
-        return {
-            "business_insights": business_insights,
-            "code_explanation": code_explanation,
-            "code_string": code_string,
-        }
+        else:
+            fig_object = _exec_code_and_retrieve_fig(
+                code=code_string, local_args={"df": df}, is_notebook_env=_is_jupyter()
+            )
+            if explain:
+                business_insights, code_explanation = self._lazy_get_component(GetCodeExplanation).run(
+                    chain_input=user_input, code_snippet=code_string
+                )
+
+                return Plot(
+                    code_string=code_string,
+                    fig=fig_object,
+                    business_insights=business_insights,
+                    code_explanation=code_explanation,
+                )
+            else:
```

Comment:
Also no need for the `else` here since the `if` clause has `return`.

Reply:
usually when there are too many if else i prefer to make them more clearly by not omitting else, but ok to delete it as well if that is the style we follow.

Reply:
sometimes i always feel lint detection is a bit mysterious, yes you are right, it should detect it by this rule:https://pylint.readthedocs.io/en/stable/user_guide/messages/refactor/no-else-raise.html? 

Recently i did not get this complaint anymore somehow